src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/UpdateDataModelBuilder.java

```diff
@@ -0,0 +1,152 @@
+
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.items.JudgementTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.items.JurCodesTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.items.RepresentedTypeRItem;
+import uk.gov.hmcts.ecm.common.model.ccd.items.RespondentSumTypeItem;
+import uk.gov.hmcts.ecm.common.model.ccd.types.*;
+import uk.gov.hmcts.ecm.common.model.multiples.MultipleData;
+import uk.gov.hmcts.ecm.common.model.servicebus.datamodel.UpdateDataModel;
+
+import java.util.List;
+import java.util.Optional;
+
+public class UpdateDataModelBuilder {
+
+    private UpdateDataModelBuilder() {
+        // Access through static methods
+    }
+
+    public static UpdateDataModel build(MultipleData multipleData, CaseData caseData) {
+        return UpdateDataModel.builder()
+                .managingOffice(multipleData.getManagingOffice())
+                .fileLocation(multipleData.getFileLocation())
+                .fileLocationGlasgow(multipleData.getFileLocationGlasgow())
+                .fileLocationAberdeen(multipleData.getFileLocationAberdeen())
+                .fileLocationDundee(multipleData.getFileLocationDundee())
+                .fileLocationEdinburgh(multipleData.getFileLocationEdinburgh())
+                .clerkResponsible(multipleData.getClerkResponsible())
+                .positionType(multipleData.getPositionType())
+                .receiptDate(multipleData.getReceiptDate())
+                .hearingStage(multipleData.getHearingStage())
+                .representativeClaimantType(getRepresentativeClaimantType(multipleData, caseData))
+                .jurCodesType(getJurCodesType(multipleData, caseData))
+                .respondentSumType(getRespondentSumType(multipleData, caseData))
+                .judgementType(getJudgementType(multipleData, caseData))
+                .representedType(getRespondentRepType(multipleData, caseData))
+                .build();
+    }
+
+    private static JurCodesType getJurCodesType(MultipleData multipleData, CaseData caseData) {
+        if (caseData == null) {
+            return null;
+        }
+
+        List<JurCodesTypeItem> jurCodesCollection = caseData.getJurCodesCollection();
+
+        if (multipleData.getBatchUpdateJurisdiction().getValue() != null
+                && jurCodesCollection != null) {
+            String jurCodeToSearch = multipleData.getBatchUpdateJurisdiction().getValue().getLabel();
+            Optional<JurCodesTypeItem> jurCodesTypeItemOptional =
+                    jurCodesCollection.stream()
+                            .filter(jurCodesTypeItem ->
+                                    jurCodesTypeItem.getValue().getJuridictionCodesList().equals(jurCodeToSearch))
+                            .findAny();
+
+            if (jurCodesTypeItemOptional.isPresent()) {
+                return jurCodesTypeItemOptional.get().getValue();
+            }
+        }
+
+        return null;
```

Comment:
My reasoning for taking this approach is that it would make the code more readable. Whilst I agree the two return null statements are not good, I prefer this to having three nested if statements.